File: ClientApp/src/app/models/subject.ts
import Assingment from './assingment';
import Course from './course';
import Enrollment from './enrollment';

export default interface Subject {
  subjectId: number;
  title: string;
  assingments: Assingment[];
  enrolments: Enrollment[];
  courseId: number;
  course: Course;
}

File: Models/Subject.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace MagnifinanceUniversity.Models
{
    public class Subject
    {
        [DatabaseGenerated(DatabaseGeneratedOption.None)]
        public int SubjectID { get; set; }
        public string Title { get; set; }
        public virtual ICollection<Assingment> Assingments { get; set; }
        public virtual ICollection<Enrollment> Enrollments { get; set; }
        public int CourseID { get; set; }
        public virtual Course Course { get; set; }
    }
}
